service/storage_service.cc

```diff
@@ -3011,14 +3016,65 @@ future<> storage_service::join_cluster(sharded<db::system_distributed_keyspace>&
         loaded_endpoints | std::views::transform([] (const auto& x) {
             return x.second.endpoint;
         }) | std::ranges::to<std::unordered_set<gms::inet_address>>();
+
+    gms::inet_address recovery_leader_ip;
+    locator::host_id recovery_leader_id;
+    if (!_db.local().get_config().recovery_leader().empty()) {
+        if (_group0->joined_group0()) {
+            // Something is wrong unless it is a noninitial (and unneeded) restart during the Raft recovery procedure.
+            slogger.warn(
+                    "recovery_leader is set to {} but persistent group 0 ID is present: {}. "
+                    "The recovery_leader option will be ignored. If you are trying to run the Raft recovery procedure, "
+                    "ensure you delete raft_group_id and truncate system.discovery.",
```

Comment:
I wouldn't include specific steps in the warning, out of context it's dangerous. Instead, point the user to the documentation, they need to read all the steps and do them in order.